web-registry/src/hooks/projects/useProjectsWithOrders.ts

```diff
@@ -0,0 +1,101 @@
+import { SellOrderInfo } from '@regen-network/api/lib/generated/regen/ecocredit/marketplace/v1/query';
+import {
+  ProjectInfo,
+  QueryProjectsResponse,
+} from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+import shuffle from 'lodash/shuffle';
+
+import {
+  ProjectsSellOrders,
+  useProjectsSellOrders,
+} from 'pages/Projects/hooks/useProjectsSellOrders';
+import useEcocreditQuery from 'hooks/useEcocreditQuery';
+import { useQuerySellOrders } from 'hooks/useQuerySellOrders';
+
+const hasMetadata = (project: ProjectInfo): boolean => !!project.metadata;
+
+const hasSellOrders = (
+  project: ProjectInfo,
+  sellOrders: SellOrderInfo[],
+): boolean =>
+  sellOrders.some(sellOrder => sellOrder?.batchDenom?.startsWith(project.id));
+
+const prioritizeWithSellOrders = (
+  projects: ProjectInfo[],
+  sellOrders: SellOrderInfo[],
+  limit?: number,
+): ProjectInfo[] => {
+  const projectsWithOrders = projects.filter(projs =>
+    hasSellOrders(projs, sellOrders),
+  );
+  if (!limit || projectsWithOrders.length >= limit) return projectsWithOrders;
+
+  const numProjects = limit - projectsWithOrders.length;
+  const additionalProjects = projects.filter(
+    p1 => !projectsWithOrders.some(p2 => p1.id === p2.id),
+  );
+  return [...projectsWithOrders, ...additionalProjects.slice(0, numProjects)];
+};
+
+const isOtherProject = (project: ProjectInfo, projectId: string): boolean =>
+  project.id !== projectId;
```

Comment:
All these utils functions should probably be in a `useProjectsWithOrders.utils.ts` file to make it more readable.